Skeletonize the project

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request skeletonizes this project from [cisagov/skeleton-python-library].
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

1. This project is in active use and has not been skeletonized.
2. I need to make changes to this project to resolve a bug and I do not want to do so without at least our linting and build verification in place.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been approved. -->

- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Add a tag or create a release.

[cisagov/skeleton-python-library](https://github.com/cisagov/skeleton-python-library

mcdonnnj committed:
Autoformat bump_version.sh with beautysh

mcdonnnj committed:
Merge pull request #68 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/add_mypy_to_dev_requirements

mcdonnnj committed:
Merge pull request #65 from cisagov/improvement/add_mypy_to_dev_requirements

Add the mypy Package to the Development Package Requirements

mcdonnnj committed:
Merge branch 'develop' into improvement/update_package_metadata

mcdonnnj committed:
Merge pull request #67 from cisagov/improvement/update_package_metadata

Update Package Metadata

mcdonnnj committed:
Update CodeQL workflow for GitHub Actions

I noticed the following warning when looking at GHA workflow runs:

1 issue was detected with this workflow: git checkout HEAD^2 is no longer
necessary. Please remove this step as Code Scanning recommends analyzing the
merge commit for best results.

After looking into it I updated the workflow per this page on GitHub Docs:

https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/troubleshooting-the-codeql-workflow#warning-git-checkout-head2-is-no-longer-necessary

mcdonnnj committed:
Merge pull request #69 from cisagov/bugfix/update_codeql_workflow

Update CodeQL Workflow for GitHub Actions

mcdonnnj committed:
Use secrets.GITHUB_TOKEN for coveralls in GHA workflow

coveralls-python has better integration with running in GitHub Actions now, so
we can use the automatically generated GITHUB_TOKEN secret instead of manually
setting a COVERALLS_REPO_TOKEN secret for each repository.

mcdonnnj committed:
Update coveralls GHA configuration for parallelization

We already ran our tests in parallel for each major version of Python a project
supports. This change should better differentiate each test job run in the
reports submitted.

mcdonnnj committed:
Skip CodeQL workflow for Dependabot branches on push events

This should resolve the following error:

Error: Workflows triggered by Dependabot on the "push" event run with read-only
access. Uploading Code Scanning results requires write access. To use Code
Scanning with Dependabot, please ensure you are using the "pull_request" event
for this workflow and avoid triggering on the "push" event for Dependabot
branches. See
https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#scanning-on-push
for more information on how to configure these events.

felddy committed:
Improve gitignore file comments and organization.

* Add a comment describing what the files does.
* Add sections to organize patterns.

hillaryj committed:
Adds a comment to direct additional requirements into setup.py

mcdonnnj committed:
Update pre-commit hooks

Hooks updated with the `pre-commit autoupdate` command.

Note:
The `ansible-lint` hook is intentionally being held back on 4.3.7 because of
ongoing issues with the 5.x version and how we use Ansible (standalone Galaxy
roles).

mcdonnnj committed:
Remove user from CODEOWNERS

Due to her departure this removes hillaryj from the default CODEOWNERS we use
in our projects.

hillaryj committed:
Merge pull request #74 from cisagov/add-req-comment

Adds a comment to direct additional requirements into setup.py

mcdonnnj committed:
Merge branch 'develop' into improvement/skip_codeql_on_push_for_dependabot_branches

mcdonnnj committed:
Merge pull request #73 from cisagov/improvement/skip_codeql_on_push_for_dependabot_branches

Skip CodeQL Workflow for Dependabot Branches on push Events

mcdonnnj committed:
Merge branch 'develop' into improvement/update_gha_coveralls_configuration

mcdonnnj committed:
Merge pull request #71 from cisagov/improvement/update_gha_coveralls_configuration

Update coveralls-python Configuration in GitHub Actions Workflow

mcdonnnj committed:
Update configuration instructions for pyenv

With the release of pyenv v2.0.0 there is a breaking change around startup
logic that necessitates an update for our pyenv setup instructions. Also
add a statement about how to get configuration instructions from pyenv
itself.

mcdonnnj committed:
Merge pull request #71 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into maintenance/remove_user_from_codeowners

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #72 from cisagov/maintenance/remove_user_from_codeowners

Remove User from CODEOWNERS

mcdonnnj committed:
Merge branch 'develop' into documentation/update_pyenv_instructions

mcdonnnj committed:
Merge pull request #75 from cisagov/documentation/update_pyenv_instructions

Update Instructions for Configuring pyenv

mcdonnnj committed:
Merge branch 'develop' into improvement/gitignore_me

mcdonnnj committed:
Merge pull request #70 from cisagov/improvement/gitignore_me

Improve gitignore file comments and organization.

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #76 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Standardize the layout of the Lineage configuration file

mcdonnnj committed:
Add to the cache keys for the GitHub Actions workflow

This adds a hash of `setup.py` to the cache keys used in the GitHub Actions
workflow for Python projects.

mcdonnnj committed:
Add comments explaining different hashFiles() argument

The other cache keys for our GHA jobs are in the format '**/<filename>' so that
any file with that name is used in the repository. However, for Python packages
they may have a 'setup.py' as part of their internal codebase that does not
impact environment requirements. As a result we only want to use the 'setup.py'
that is in the root of the repository and is used to install the package.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Add style enforcement rules

Add rules to enforce ATX-closed headers, dashes for unordered list elements,
and `1.` for ordered list elements.

mcdonnnj committed:
Add rule for image headers

Add <h1> and <img> tags to the allowed list for MD033 (HTML elements) to
support using an image as the first thing in a markdown file (header image).

mcdonnnj committed:
Switch to a YAML markdownlint configuration file

This converts the existing `.mdl_config.json` file to an equivalent
`.mdl_config.yaml` file. The reference in the markdownlint pre-commit hook
configuration is updated to match.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add comments to markdownlint configuration

Now that this is a YAML file we can add comments explaining the rule
modifications we use. This will make it easier to edit or expand in the future.

mcdonnnj committed:
Add the validate_manifest hook from pre-commit

This hook will validate any pre-commit hook manifest files in the repository.

jsf9k committed:
Add assertion to verify that the root logger's logging level is set correctly

See the discussion here for more context:
https://github.com/cisagov/pe-reports/pull/6#discussion_r657329612

jsf9k committed:
Install terraform and packer for the linting job

We should be doing this because the Packer and Terraform pre-commit
hooks leverage the corresponding executables; therefore, it makes
sense to go ahead and install the particular versions of those
executables that we support.  Also add support for optionally
debugging via tmate.

See also cisagov/skeleton-generic#74.

jsf9k committed:
Break out the curl cache creation into its own step

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove unnecessary line in tasks

There is no reason to create /usr/bin/terraform.  This is a vestige of
an earlier age.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Modify the Packer installation to model that of Terraform

The Terraform installation does not destroy the existing system
Terraform installation, and neither should the Packer installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Install a specific version of terraform-docs

Note that this change is dependent on the merging of
cisagov/setup-env-github-action#31.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Move go installation so that it takes place before the cache task

Some variables defined in the go installation are used in the cache
task, so the go installation must happen first.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Capitalize Go for consistency

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Update pre-commit hooks

This is performed by running `pre-commit autoupdate`, but with the
`ansible-lint` hook held back manually.

jsf9k committed:
Prefer the newer "go install" syntax

As of [Go 1.16](https://tip.golang.org/doc/go1.16#go-command) the `GO111MODULE` environment variable defaults to `on` and `go get` has been deprecated for module installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Install the shfmt tool for GHA

The `shfmt` tool does not ship on the GitHub Actions runners so we must install
it manually.

mcdonnnj committed:
Replace the beautysh hook with pre-commit-shfmt

We have had a difficult time with how beautysh parses some shellscripts. I went
in pursuit of an alternative and I believe shfmt to be a good alternative.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Apply changes from the shfmt pre-commit hook

mcdonnnj committed:
Remove unnecessary boilerplate

This package has a `__main__.py` file which removes the need to include this
boilerplate.

mcdonnnj committed:
Explicitly call sys.exit for errors

Instead of returning a value the main loop should explicitly call `sys.exit()`
if an error is encountered that would result in a non-zero exit code.

mcdonnnj committed:
Update function type hints

The type hints for the example_div function indicate it takes two float values,
but the command line interface indicates (and checks) that it takes integer
values.

mcdonnnj committed:
Do not use f-strings in logging messages

C-style '%' formatting should be used to any calls to the `logging` library.

mcdonnnj committed:
Update testing for exit code changes

The testing needs to be updated now that `example.main()` only returns a
`None` and error exits are done by directly calling `sys.exit()`.

mcdonnnj committed:
Merge pull request #83 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/add_pre-commit_hook

mcdonnnj committed:
Merge pull request #80 from cisagov/improvement/add_pre-commit_hook

Add the validate_manifest Hook to pre-commit Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/update_mdl_configuration

jsf9k committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #79 from cisagov/improvement/update_mdl_configuration

Update markdownlint Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #82 from cisagov/improvement/install-tf-and-packer-for-linting

Install terraform and packer for the linting job

mcdonnnj committed:
Merge branch 'develop' into improvement/replace_beautysh_with_shfmt

mcdonnnj committed:
Merge pull request #84 from cisagov/improvement/replace_beautysh_with_shfmt

Replace beautysh hook with cisagov/pre-commit-shfmt

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Changes from the pre-commit-shfmt hook

mcdonnnj committed:
Add stubs required by mypy pre-commit hook

With the release of 0.900 mypy moved to a modular stub model. This means that
only stdlib stubs are "baked in" and everything else must be installed
separately. Since the mypy hook runs in a venv the stubs it recommends need to
be installed as additional dependencies.

jsf9k committed:
Install mypy type stubs as part of setup-env

We already do this as part of the mypy pre-commit hook, but this way
mypy is also ready to run manually in the developer's local Python
virtual env.

mcdonnnj committed:
Merge pull request #82 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/standardize_lineage_configuration

mcdonnnj committed:
Merge pull request #77 from cisagov/improvement/standardize_lineage_configuration

Standardize the Layout of the Lineage Configuration File

mcdonnnj committed:
Merge branch 'develop' into improvement/improve-test

mcdonnnj committed:
Merge pull request #80 from cisagov/improvement/improve-test

Verify that the logging level for the root logger is set correctly

mcdonnnj committed:
Merge branch 'develop' into improvement/changes_to_match_development_preferences

mcdonnnj committed:
Merge pull request #81 from cisagov/improvement/changes_to_match_development_preferences

Update to Match Best Practices

mcdonnnj committed:
Merge branch 'develop' into improvement/add_setup_to_gha_cache_key

mcdonnnj committed:
Merge pull request #78 from cisagov/improvement/add_setup_to_gha_cache_key

Add setup.py to the Cache Keys for the GitHub Actions Workflow

mcdonnnj committed:
Switch to block-style sequence in build workflow

Switch from a flow style to a block style sequence in the build workflow. This
will improve maintainability by making version changes impact one Python
version at a time.

dav3r committed:
Fix a typo

mcdonnnj committed:
Use the hashicorp/setup-terraform Action

Instead of manually installing a Terraform binary we can leverage the Action
provided by Hashicorp to do the same thing.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Add a markdownlint rule for horizontal rules

This sets a specific rule for MD035 (Horizontal rule style) instead of the
default value of "consistent".

mcdonnnj committed:
Add a markdownlint rule for code blocks

This sets a specific rule for MD046 (code block style) instead of the default
value of "consistent".

mcdonnnj committed:
Merge branch 'develop' into dav3r-patch-1

mcdonnnj committed:
Merge pull request #85 from cisagov/dav3r-patch-1

Fix a typo

mcdonnnj committed:
Merge branch 'develop' into improvement/specify_mdl_rules

mcdonnnj committed:
Merge pull request #89 from cisagov/improvement/specify_mdl_rules

Add markdownlint Rules to Enforce Specific Formatting

mcdonnnj committed:
Merge branch 'develop' into improvement/use_setup-terraform_in_gha_workflow

mcdonnnj committed:
Merge pull request #88 from cisagov/improvement/use_setup-terraform_in_gha_workflow

Use the hashicorp/setup-terraform Action to Install Terraform in GHA Workflow

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

dav3r committed:
Enable terraform validate pre-commit hook

This hook should now work as expected in most, if not all of our repos 
now that we are finally updating to terraform 0.13.x (on our way to 
1.0.x).

mcdonnnj committed:
Merge pull request #85 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/gha_python_version_block_style_sequence

mcdonnnj committed:
Merge pull request #83 from cisagov/improvement/gha_python_version_block_style_sequence

Switch to Block-Style Sequence for Python Versions in build Workflow

dav3r committed:
Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13

This change will be reverted when testing is completed.

dav3r committed:
Remove lint job step to initialize Terraform directories

Initialization will now be done during the "terraform validate" step.

dav3r committed:
Revert "Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13"

This reverts commit 6a7fbf07bd371d0493c523ce24647e5c04c77c03.

dav3r committed:
Merge pull request #90 from cisagov/improvement/enable_terraform_validate

Enable terraform validate pre-commit hook

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Use an id when using cisagov/setup-env-github-action

mcdonnnj committed:
Use setup-env outputs for Terraform version

Change from using an environment variable to the outputs of the
cisagov/setup-env-github-action instead.

mcdonnnj committed:
Use setup-env outputs for Packer version

We use a `PACKER_VERSION` environment variable for the "Install Packer" step
that is populated from the `setup-env` outputs to get around `yamllint` lint
length limits.

mcdonnnj committed:
Use setup-env outputs for shfmt version

We change the "Install shfmt" step to use two local environment variables to
provide the package's URL and version to install. This allows us to work around
`yamllint` line length limits.

mcdonnnj committed:
Use setup-env outputs for Terraform-docs version

We change the "Install Terraform-docs" step to use two local environment
variables to provide the package's URL and version to install. This allows us
to work around `yamllint` line length limits.

mcdonnnj committed:
Store Go version as a step output

This changes from using an environment variable to using a step output to store
the Go version that is installed. This mirrors changes made to the other
program versions and how they're stored.

mcdonnnj committed:
Update the ansible-lint version for pre-commit

mcdonnnj committed:
Add an ansible-lint configuration file

This file is copied from cisagov/skeleton-ansible-role#85 and tweaked to pass
pre-commit. See here for for a list of the elements that can exist in this
file: https://ansible-lint.readthedocs.io/en/latest/configuring.html

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add Python 3.10 to GHA version matrices

Add Python 3.10 to the matrices of versions used by the `test` and `build`
steps.

mcdonnnj committed:
Force strings for GHA version matrices

Ensure that the versions listed in the version matrices used by the `test` and
`build` jobs are interpreted as strings. Without this change the version `3.10`
is interpreted as `3.1` when converted to a fixnum. Since these are intended to
be interpreted as string values we should enforce it for all versions listed.

mcdonnnj committed:
Add dependabot configuration for Terraform

This will configure `dependabot` to scan Terraform configurations if they exist
in a repository. If a repository stores a Terraform configuration in a sub-
directory this configuration will need to be modified or an additional
configuration must be added if there still exists a Terraform configuration in
the root directory.

mcdonnnj committed:
Merge pull request #91 from cisagov/improvement/pre-commit_update

Update `pre-commit` Hooks

mcdonnnj committed:
Merge pull request #93 from cisagov/improvement/update_ansible-lint_pre-commit_version

Update the `ansible-lint` Version in the pre-commit Configuration

mcdonnnj committed:
Merge pull request #94 from cisagov/improvement/add_dependabot_configuration_for_terraform

Add `dependabot` Configuration for Terraform

mcdonnnj committed:
Merge pull request #92 from cisagov/improvement/use_action_outputs

Use Outputs of cisagov/setup-env-github-action to Control Installed Program Versions

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Update ansible-lint configuration

Remove the manual exclusion of the `converge.yml` file in the default molecule
scenario. With the changes in cisagov/skeleton-ansible-role#88 this exclusion
is no longer necessary.

mcdonnnj committed:
Update ansible-lint configuration

Instead of excluding molecule configurations for systemd individually we
instead direct ansible-lint to process any of these configurations in
scenario directories as plain yaml files. This mirrors how ansible-lint
handles `molecule.yml` files in scenario directories by default.

mcdonnnj committed:
Merge pull request #86 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #87 from cisagov/improvement/add_support_for_py3.10

Add Support for Python 3.10

mcdonnnj committed:
Disable fail-fast for matrix GHA jobs

Override the default by setting `fail-fast` to `false` for the test and build
jobs in our GHA build workflow. This ensure that all matrix jobs run regardless
of a failure.

mcdonnnj committed:
Merge pull request #88 from cisagov/improvement/disable_fail-fast_for_test_and_build_jobs

Disable `fail-fast` for GitHub Actions Jobs with a Matrix Strategy

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

mcdonnnj committed:
Merge pull request #97 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/update_ansible-lint_configuration

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/update_ansible-lint_configuration

Update `ansible-lint` Configuration

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #89 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Add Python 3.10 to package classifiers

This was missed when we added testing/building for Python 3.10.

mcdonnnj committed:
Bump version from 0.0.1 to 0.1.0

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

mcdonnnj committed:
Add comment to Dependabot configuration

This comment explains that the configuration may have commented out
ignore directives that should be uncommented in downstream projects.

mcdonnnj committed:
Disable comments-indentation rule for yamllint

yamllint does not like it when you comment out pieces of dictionaries
in lists. Upcoming additions to the Dependabot configuration will run
afoul of this so we are updating the yamllint configuration.

mcdonnnj committed:
Merge pull request #99 from cisagov/improvement/update_dependabot_and_yamllint_configurations

Add boilerplate for upcoming Dependabot configuration changes

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #98 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit hooks

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Use asterisks for emphasis- and strong-styles

Given the inconsistent behavior of underscores used within words for
style we should prefer asterisks now that these rules are available.

mcdonnnj committed:
Add missing hyphen in markdownlint rule header

mcdonnnj committed:
Merge pull request #100 from cisagov/improvement/update_markdownlint_configuration

Use new `markdownlint` rules for emphasis- and strong-styles

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

dependabot[bot] committed:
Bump actions/setup-python from 2 to 3

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 2 to 3.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #91 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/add_official_py3.10_support

mcdonnnj committed:
Merge pull request #93 from cisagov/improvement/add_official_py3.10_support

Officially support Python 3.10

mcdonnnj committed:
Update git ignore list

Add dist/ to the .gitignore file to better support local building.

mcdonnnj committed:
Modernize the Python package build step

Modernize the build process by using the build package instead of the
now deprecated setup.py install method. Remove installing the package
requirements as it is unnecessary for this build process. Upgrade the
setuptools package as part of the initial dependency installation.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with upgrading to v6.

mcdonnnj committed:
Add Dependabot ignore directives

This adds commented out ignore directives for the following GitHub
Actions:

- action/cache
- action/checkout
- action/setup-python

These should be uncommented downstream to ensure that updates to these
dependencies are pushed from pull requests made in the skeleton.

mcdonnnj committed:
Upgrade from Python 3.9 to 3.10 for the `lint` job in GHA

mcdonnnj committed:
Use consistent quoting for software versions

mcdonnnj committed:
Merge pull request #101 from cisagov/dependabot/github_actions/actions/setup-python-3

Bump actions/setup-python from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into dependabot/github_actions/actions/checkout-3

mcdonnnj committed:
Merge pull request #102 from cisagov/dependabot/github_actions/actions/checkout-3

Bump actions/checkout from 2 to 3

dependabot[bot] committed:
Bump actions/cache from 2 to 3

Bumps [actions/cache](https://github.com/actions/cache) from 2 to 3.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge branch 'develop' into improvement/add_dependabot_ignores_for_github_actions

mcdonnnj committed:
Merge pull request #107 from cisagov/improvement/add_dependabot_ignores_for_github_actions

Add Dependabot ignore directives

mcdonnnj committed:
Merge pull request #104 from cisagov/dependabot/github_actions/actions/cache-3

Bump actions/cache from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into improvement/use_python_3.10_for_gha

mcdonnnj committed:
Merge pull request #106 from cisagov/improvement/use_python_3.10_for_gha

Use Python 3.10 for the `lint` job in GitHub Actions

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #105 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Enable Dependabot ignore directives

This enables the Dependabot ignore directives for GitHub Actions.

mcdonnnj committed:
Add permission scoping for the CodeQL Analyze GHA job

This sets permissions around `GITHUB_TOKEN` to the minimum necessary
for this workflow. Please see
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
for more information.

mcdonnnj committed:
Update github/codeql-action/init from 1 to 2

Update the comments to better mirror the current documentation for
github/codeql-action. Switch to block style for the map of languages.

mcdonnnj committed:
Consistently use block style in the CodeQL GHA workflow

mcdonnnj committed:
Update github/codeql-action/autobuild from 1 to 2

Clean up the commenting while we're here.

mcdonnnj committed:
Update github/codeql-action/analyze from 1 to 2

Also clean up the commenting while we're here.

mcdonnnj committed:
Add pre-commit hooks to validate GitHub Actions content

Add hooks to validate GitHub Actions and GitHub Action workflows. These
are validated using JSON Schemas.

mcdonnnj committed:
Merge pull request #109 from cisagov/improvement/add_github_actions_pre-commit_hooks

Add `pre-commit` hooks to validate GitHub Actions content

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #97 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #98 from cisagov/improvement/update_codeql_workflow

Update the CodeQL workflow

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/update_build_process

Modernize the `build` step in the `build` workflow

mcdonnnj committed:
Add an ownership comment to the Dependabot configuration

Add a comment that states that the commented out ignore directives are
managed by cisagov/skeleton-generic.

mcdonnnj committed:
Add additional Dependabot ignore directives

This adds the other versioned Actions that should be managed by
cisagov/skeleton-generic to the list of commented out dependencies to
ignore.

mcdonnnj committed:
Update actions/setup-go from v2 to v3

Additionally as of v3.1.0 of actions/setup-go there is a go-version
output value to retrieve the version of Go installed by the Action.
This allows us to remove the step to manually retrieve this information
from the Go executable.

mcdonnnj committed:
Update the version of Go used from 1.16 to 1.18

Go 1.16 is no longer supported as of the release of 1.18 so it makes
sense to update to the latest version available.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with upgrading to v6.

mcdonnnj committed:
Enable the allow-non-breakable-inline-mappings rule for yamllint

mcdonnnj committed:
Increase the maximum line length in the yamllint configuration

Increase the line length maximum from the default of 80 to 88. This 10%
increase mirrors the line length allowed in other linter configurations
such as black.

jsf9k committed:
Fix overly match-happy sed commands

We saw in cisagov/postfix-docker#47 that the sed commands in the
bump_version.sh script could inadvertently match the CC0 version in
the README.md file.  This change escapes the periods in the version
before passing it on to sed so that they only match periods and not
just any character.

mcdonnnj committed:
Add a label configuration file

This configuration file stores information about the labels expected in
this repository.

mcdonnnj committed:
Add a workflow to synchronize repository labels

This adds a workflow to ensure that the repository labels are updated
to reflect changes to the label configuration file .github/labels.yml.

mcdonnnj committed:
Remove usage of set-output from our GitHub Actions workflow

GitHub has deprecated the set-output command per:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

This updates the GitHub Actions workflow to use the newly preferred
method to set the output for a job's step.

mcdonnnj committed:
Fix the repo of the flake8 pre-commit hook

Per PyCQA/flake8#1290 this hook moved from GitLab to GitHub. The
version we use is bumped to the latest tag on GitHub as well.

jsf9k committed:
Remove duplicate label

jsf9k committed:
Add a label for issues or PRs that involve test code

mcdonnnj committed:
Add support for Python 3.11

With the release of Python 3.11 on 2022-10-24 we should add it to our
supported Python versions. This includes an appropriate PyPI trove
classifier and adding it to the Python versions that are tested in our
GitHub Actions workflow.

mcdonnnj committed:
Add additional PyPI trove classifiers

Since we only support Python 3 and only test against CPython (the
reference implementation) it would be helpful to indicate these facts
with our trove classifiers.

mcdonnnj committed:
Update the version of Go used from 1.18 to 1.19

Go 1.19 was released while this branch was in the wings and it makes
sense to bump to the latest Go release.

mcdonnnj committed:
Merge pull request #113 from cisagov/improvement/update_go_configuration

Update Go installation in the `build.yml` workflow

dependabot[bot] committed:
Bump hashicorp/setup-terraform from 1 to 2

Bumps [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform) from 1 to 2.
- [Release notes](https://github.com/hashicorp/setup-terraform/releases)
- [Changelog](https://github.com/hashicorp/setup-terraform/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/setup-terraform/compare/v1...v2)

---
updated-dependencies:
- dependency-name: hashicorp/setup-terraform
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #111 from cisagov/dependabot/github_actions/hashicorp/setup-terraform-2

Bump hashicorp/setup-terraform from 1 to 2

dependabot[bot] committed:
Bump actions/setup-python from 3 to 4

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 3 to 4.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #115 from cisagov/dependabot/github_actions/actions/setup-python-4

Bump actions/setup-python from 3 to 4

mcdonnnj committed:
Merge pull request #112 from cisagov/improvement/update_dependabot_ignores

Update Dependabot ignore directives

mcdonnnj committed:
Merge pull request #119 from cisagov/bug/update_github_actions_workflow

Remove usage of `set-output` from our GitHub Actions workflow

mcdonnnj committed:
Merge pull request #117 from cisagov/improvement/update_yamllint_line-length_rules

Update the `line-length` configuration for `yamllint`

mcdonnnj committed:
Merge pull request #120 from cisagov/bug/verify_pre-commit_hook_repos

Update the `flake8` pre-commit hook configuration

mcdonnnj committed:
Merge pull request #114 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit hooks

mcdonnnj committed:
Merge pull request #118 from cisagov/improvement/add_label_sync_workflow

Add workflow to manage repository labels

mcdonnnj committed:
Fix GitHub label color values

Update the configuration for repository labels to remove the leading
`#` from color values. With a `#` leading the values they are seen as
invalid by the GitHub API.

mcdonnnj committed:
Correct spacing in yamllint rules

There was a missing empty line in the `.yamllint` file between two rule
definitions.

mcdonnnj committed:
Merge pull request #122 from cisagov/bug/fix_bad_configurations

Update two configuration files

mcdonnnj committed:
Update the version of the second bandit hook

The `bandit (everything else)` hook was not updated in sync with the
`bandit (tests tree)` hook and is using an older version of bandit.

mcdonnnj committed:
Correct poor English in comments for the bandit pre-commit hooks

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Add a CodeQL status badge to the README

mcdonnnj committed:
Pin Python 3.6 testing to Ubuntu 20.04

There is no Python 3.6 on Ubuntu 22.04 release available from
actions/python-versions. Since Ubuntu 22.04 is what is being used for
the `ubuntu-latest` tag we need to pin Python 3.6 testing to use the
old `ubuntu-20.04` tag.

mcdonnnj committed:
Remove LGTM badges from the README

Now that LGTM has been deprecated in favor of GitHub code scanning
(CodeQL) we have removed the integration from the organization. As a
result we need to remove these badges.

mcdonnnj committed:
Add a GitHub Actions job to test wheels

This job will test thw wheels created during the `build` job to ensure
the distribution artifact also passes tests.

mcdonnnj committed:
Prefer block-style to flow-style in GitHub Actions job needs

mcdonnnj committed:
Merge pull request #114 from cisagov/bug/pin_ubuntu_image_for_py3.6

Pin Python 3.6 testing to Ubuntu 20.04

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Add a `python` label

Since this project is for a Python package it makes sense to include a
`python` label.

mcdonnnj committed:
Merge pull request #110 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #111 from cisagov/bug/update_bandit_in_pre-commit_configuration

Update the version of the second `bandit` hook

mcdonnnj committed:
Merge pull request #115 from cisagov/improvement/update_codeql_workflow

Add a CodeQL badge and remove LGTM badges from the README

mcdonnnj committed:
Merge pull request #108 from cisagov/improvement/update_trove_classifiers

Add two additional PyPI trove classifiers

mcdonnnj committed:
Merge pull request #104 from cisagov/bugfix/tighten-regex-in-version-script

Fix overly match-happy sed commands

mcdonnnj committed:
Merge pull request #116 from cisagov/improvement/test_wheels_built_in_github_actions

Add a GitHub Actions job to test built wheels

mcdonnnj committed:
Bump version from 0.1.0 to 0.2.0

mcdonnnj committed:
Merge pull request #109 from cisagov/improvement/add_support_for_python_3.11

Add support for Python 3.11

dependabot[bot] committed:
Bump actions/upload-artifact from 2 to 3

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 2 to 3.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Add ignore directives to the dependabot configuration

Add ignore directives and an attribution comment to the dependabot
configurtation. These will be uncommented in descendants of this
project (cisagov/skeleton-python-library) so that version changes to
these Actions are controlled through this project.

mcdonnnj committed:
Merge pull request #102 from cisagov/dependabot/github_actions/actions/upload-artifact-3

Bump actions/upload-artifact from 2 to 3

mcdonnnj committed:
Merge pull request #117 from cisagov/improvement/add_github_actions_attributions

Add ignore directives to the dependabot configuration

mcdonnnj committed:
Merge github.com:cisagov/skeleton-python-library into skeletonize

mcdonnnj committed:
Convert the trustymail script into a sub-module

Bring the command line functionality directly into the package and use
an entry point instead of a script.

mcdonnnj committed:
Remove TravisCI configuration

We no longer use TravisCI so this should be removed.

mcdonnnj committed:
Remove configurations that do not align with the skeleton

The cisagov/skeleton-python-library project does not use MANIFEST.in or
setup.cfg. It also does not use Tox. Therefore all of these
configuration files should be removed for the sake of consistency.

mcdonnnj committed:
Remove execute bit from the package's Python files

These files should not be executable.

mcdonnnj committed:
Update the Dockerfile

Update the Dockerfile to account for changes to the package structure.

mcdonnnj committed:
Apply changes from the `pyupgrade` hook

Add the changes made from running the `pyupgrade` pre-commit hook
against the project.

mcdonnnj committed:
Apply changes from the `isort` hook

Add the changes made from running the `isort` pre-commit hook against
the project.

mcdonnnj committed:
Update to pass the `markdownlint` hook

Make the necessary changes to pass the `markdownlint` pre-commit hook.

mcdonnnj committed:
Update to pass the `mypy` hook

Make the necessary changes to pass the `mypy` pre-commit hook.

mcdonnnj committed:
Update to pass the `flake8` hook

Make the necessary changes to pass the `flake8` pre-commit hook. This
is almost exclusively to satisfy the flake8-docstring plugin.

mcdonnnj committed:
Update references to reflect this project

Update the Lineage parent to cisagov/skeleton-python-library and
replace any references to cisagov/skeleton-python-library with this
project.

mcdonnnj committed:
Enable dependabot ignore directives

Enable all of the dependabot ignore directives managed by the
cisagov/skeleton-python-library project.

mcdonnnj committed:
Bump version from 0.7.5 to 0.8.0

mcdonnnj committed:
Bump version from 0.8.0 to 0.8.0-rc.1

dav3r commented on .coveragerc:
Replace skeleton directory name:
```suggestion
source = src/trustymail
```

dav3r commented on src/trustymail/trustymail.py:
Fix a typo while we are here:
```suggestion
    Parse a DMARC Reporting (i.e. ``rua``/``ruf``) URI.
```

dav3r reviewed (approved):
Looks great aside from the two minor items I noted.  Strong work! 🚀

mcdonnnj commented on .coveragerc:
Ack I thought I'd changed that.

mcdonnnj committed:
Update coverage configuration

Use the correct package source for coverage information that is
generated and sent to Coveralls.

Co-authored-by: dav3r <daver@geekpad.com>

mcdonnnj committed:
Fix typo in README

Co-authored-by: dav3r <daver@geekpad.com>

mcdonnnj committed:
Finalize version from 0.8.0-rc.1 to 0.8.0
